velox/substrait/VeloxToSubstraitPlan.cpp

```diff
@@ -0,0 +1,158 @@
+
+::substrait::Plan* VeloxToSubstraitPlanConvertor::toSubstrait(
+    google::protobuf::Arena& arena,
+    const std::shared_ptr<const PlanNode>& plan) {
+  // Assume only accepts a single plan fragment.
+  // TODO add root_rel
+  ::substrait::Plan* substraitPlan =
+      google::protobuf::Arena::CreateMessage<::substrait::Plan>(&arena);
+  ::substrait::Rel* rel = substraitPlan->add_relations()->mutable_rel();
+  toSubstrait(arena, plan, rel);
+
+  return substraitPlan;
+}
+
+void VeloxToSubstraitPlanConvertor::toSubstrait(
+    google::protobuf::Arena& arena,
+    const std::shared_ptr<const PlanNode>& planNode,
+    ::substrait::Rel* rel) {
+  if (auto filterNode = std::dynamic_pointer_cast<const FilterNode>(planNode)) {
+    auto filterRel = rel->mutable_filter();
+    toSubstrait(arena, filterNode, filterRel);
+    return;
+  }
+  if (auto valuesNode = std::dynamic_pointer_cast<const ValuesNode>(planNode)) {
+    ::substrait::ReadRel* readRel = rel->mutable_read();
+    toSubstrait(arena, valuesNode, readRel);
+    return;
+  }
+  if (auto projectNode =
+          std::dynamic_pointer_cast<const ProjectNode>(planNode)) {
+    ::substrait::ProjectRel* projectRel = rel->mutable_project();
+    toSubstrait(arena, projectNode, projectRel);
+    return;
+  }
+}
+
+void VeloxToSubstraitPlanConvertor::toSubstrait(
+    google::protobuf::Arena& arena,
+    const std::shared_ptr<const FilterNode>& filterNode,
+    ::substrait::FilterRel* filterRel) {
+  std::vector<std::shared_ptr<const PlanNode>> sources = filterNode->sources();
+
+  // Check there only have one input.
+  VELOX_USER_CHECK_EQ(
+      1, sources.size(), "Filter plan node must have extract one source.");
+  auto source = sources[0];
+
+  ::substrait::Rel* filterInput = filterRel->mutable_input();
+  // Build source.
+  toSubstrait(arena, source, filterInput);
+
+  // Construct substrait expr(Filter condition).
+  auto filterCondition = filterNode->filter();
+  auto inputType = source->outputType();
+  filterRel->set_allocated_condition(
+      exprConvertor_.toSubstraitExpr(arena, filterCondition, inputType));
+
+  filterRel->mutable_common()->mutable_direct();
+}
+
+void VeloxToSubstraitPlanConvertor::toSubstrait(
+    google::protobuf::Arena& arena,
+    const std::shared_ptr<const ValuesNode>& valuesNode,
+    ::substrait::ReadRel* readRel) {
+  const auto& outputType = valuesNode->outputType();
+
+  ::substrait::ReadRel_VirtualTable* virtualTable =
+      readRel->mutable_virtual_table();
+
+  readRel->set_allocated_base_schema(
+      typeConvertor_.toSubstraitNamedStruct(arena, outputType));
+  // The row number of the input data.
+  int64_t numRows = valuesNode->values().size();
+
+  // There have multiple rows in the data and each row is a RowVectorPtr.
+  for (int64_t row = 0; row < numRows; ++row) {
+    // The row data.
+    ::substrait::Expression_Literal_Struct* litValue =
+        virtualTable->add_values();
+    const auto& rowValue = valuesNode->values().at(row);
```

Comment:
rowValue is a RowVector, not a single value, hence, rowVector might be a better name

Reply:
change this.